Assets/Code/VehicleEditor/EditSystem.cs

```diff
@@ -307,4 +375,168 @@ partial class EditSystem : SystemBase {
+
+                var parentPart = siblings_of_current_parent[j].Value;
+                if (EntityManager.Exists(parentPart))
+                {
+
+                    if ((depth + 1) == iteration)
+                    { 
+                        var = false;
+                        RotatingPartPlacement(ref actionQueue, part: part, parentPart: parentPart, firstPartPos: firstPartPos, symmetryCount: symmetryCount, resetlayer: false, placeTransform: placeTransform);
+                    }
+                    else
+                    {
+                        RecursifFor(depth: depth + 1, parentPart);
+                    }
+                }
+                // THE IMPORTANT PART : ROTATING // Ok I could use the placeTransform as first part pos but when I first created this program I wanted to make sure that position and rotation dont interfere 
+                firstPartPos = (rotate * (firstPartPos - center)) + (Vector3)center;
+                placeTransform.Rotation = (Quaternion)placeTransform.Rotation * rotate;
+            }
+        }
+        RecursifFor(depth:0, lastParent:entitiesRootToBottom[0]);
+
+    }
+
+    void RotatingPartPlacement(ref EntityCommandBuffer actionQueue, Entity part, Entity parentPart, float3 firstPartPos, int symmetryCount, bool resetlayer, LocalTransform placeTransform) {
+        float3 center = SystemAPI.GetComponent<LocalTransform>(parentPart).Position;
+
+        Entity[] siblings = new Entity[symmetryCount];
+        for (int i = 0; i < symmetryCount; i++)
+        {
+            Quaternion rotation = Quaternion.AngleAxis(i * 360 / symmetryCount, Vector3.up);
+            float3 p = (rotation * (firstPartPos - center)) + (Vector3)center;
+            //p.y += 0.25f * i; // debuging purpsers shows which part is placed first second thrid ...
+            LocalTransform newp = new LocalTransform
+            { Position = p, Rotation = placeTransform.Rotation * rotation, Scale = 1 };
+            siblings[i] = place_parts(ref actionQueue, partToPlace: part, parentEntity: parentPart, placementTransform: newp);
+        }
+
+        // Set Siblings to every siblings // And Part
+        int layer;
+        if (!resetlayer) { layer = SystemAPI.GetComponent<Part>(parentPart).layer + 1; }
+        else layer = 1;
+        for (int i = 0; i < symmetryCount; i++)
+        { 
+            actionQueue.AddBuffer<Siblings>(siblings[i]);
+            // THIS "PART" is stored a lot of times with the same value , if need to optimse change the structre
+            actionQueue.AddComponent(siblings[i], new Part { id = lastID+1 , layer = layer , selfplace = i}); // this lastId +1  is done a lot of time , it dont matter but if realy need to optimse create a current Id and add1 each time before creating parts
+
+            for (int j = 0; j < symmetryCount; j++) { actionQueue.AppendToBuffer(siblings[i], new Siblings { Value = siblings[j] }); }
+            
+        }
+    }
+}
+
+public struct Parente : IComponentData
```

Comment:
put component structs in their own files (or, better yet, put their functions in existing component structs)

Reply:
ok